browser/ai_chat/tools/tab_management_tool.cc

```diff
@@ -0,0 +1,1378 @@
+    HandleListTabs(std::move(callback));
+  } else if (*action == "move") {
+    HandleMoveTabs(std::move(callback), dict);
+  } else if (*action == "close") {
+    HandleCloseTabs(std::move(callback), dict);
+  } else if (*action == "create_group") {
+    HandleCreateGroup(std::move(callback), dict);
+  } else if (*action == "update_group") {
+    HandleUpdateGroup(std::move(callback), dict);
+  } else if (*action == "remove_from_group") {
+    HandleRemoveFromGroup(std::move(callback), dict);
+  } else {
+    std::move(callback).Run(CreateContentBlocksForText(
+        "Invalid action. Must be one of: list, move, close, "
+        "create_group, update_group, remove_from_group"));
+  }
+}
+
+base::Value::Dict TabManagementTool::GenerateTabList() const {
+  base::Value::Dict result;
+  base::Value::List windows;
+
+  // Iterate through all browser windows for this profile
+  for (Browser* browser : *BrowserList::GetInstance()) {
+    if (browser->profile() != profile_) {
+      continue;
+    }
+
+    base::Value::Dict window_info;
+    window_info.Set("window_id", browser->session_id().id());
+    window_info.Set("is_active", browser->window()->IsActive());
+
+    TabStripModel* tab_strip = browser->tab_strip_model();
+    if (!tab_strip) {
+      continue;
+    }
+
+    base::Value::List tabs;
+    base::Value::Dict groups;
+
+    // First, collect group information
+    if (tab_strip->SupportsTabGroups() && tab_strip->group_model()) {
+      for (const auto& group_id : tab_strip->group_model()->ListTabGroups()) {
+        const TabGroup* group = tab_strip->group_model()->GetTabGroup(group_id);
+        if (group) {
+          base::Value::Dict group_info;
+          const auto* visual_data = group->visual_data();
+          group_info.Set("title", base::UTF16ToUTF8(visual_data->title()));
+          group_info.Set("color", GetTabGroupColorString(visual_data->color()));
+          group_info.Set("is_collapsed", tab_strip->IsGroupCollapsed(group_id));
+
+          groups.Set(group_id.ToString(), std::move(group_info));
+        }
+      }
+    }
+
+    // Collect tab information
+    for (int i = 0; i < tab_strip->count(); ++i) {
+      content::WebContents* web_contents = tab_strip->GetWebContentsAt(i);
+      if (!web_contents) {
+        continue;
+      }
+
+      base::Value::Dict tab_info;
+      // Use TabHandle instead of extension tab ID
+      if (auto* tab_interface =
+              tabs::TabInterface::GetFromContents(web_contents)) {
+        tab_info.Set("tab_id", tab_interface->GetHandle().raw_value());
+      } else {
+        // Fallback - this shouldn't happen for normal tabs
+        tab_info.Set("tab_id", -1);
+      }
+      tab_info.Set("index", i);
+      tab_info.Set("url", web_contents->GetURL().spec());
+      tab_info.Set("title", base::UTF16ToUTF8(web_contents->GetTitle()));
+      tab_info.Set("is_active", i == tab_strip->active_index());
+      tab_info.Set("is_pinned", tab_strip->IsTabPinned(i));
```

Comment:
unfortunately you'll need to track these other properties too

Reply:
Moved everything to BrowserWindowInterface and TabHandle